pkg/objstore/s3/s3.go

```diff
@@ -40,13 +41,20 @@ const DirDelim = "/"
 
 // Config stores the configuration for s3 bucket.
 type Config struct {
-	Bucket        string `yaml:"bucket"`
-	Endpoint      string `yaml:"endpoint"`
-	AccessKey     string `yaml:"access-key"`
-	Insecure      bool   `yaml:"insecure"`
-	SignatureV2   bool   `yaml:"signature-version2"`
-	SSEEncryption bool   `yaml:"encrypt-sse"`
-	SecretKey     string `yaml:"secret-key"`
+	Bucket                string         `yaml:"bucket"`
+	Endpoint              string         `yaml:"endpoint"`
+	AccessKey             string         `yaml:"access-key"`
+	Insecure              bool           `yaml:"insecure"`
+	SignatureV2           bool           `yaml:"signature-version2"`
+	SSEEncryption         bool           `yaml:"encrypt-sse"`
+	SecretKey             string         `yaml:"secret-key"`
+	IdleConnTimeout       model.Duration `yaml:"idle-conn-timeout"`
```

Comment:
Can we do yaml names with underscore? I don't know how we slipped the `dash` convention, but I would stay with the style Prometheus has: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#%3Cscrape_config%3E